compiler/unittest_light.nim

```diff
@@ -16,20 +16,19 @@ proc mismatch*[T](lhs: T, rhs: T): string =
   result.add "\n"
   result.add "lhs:{" & replaceInvisible(
       $lhs) & "}\nrhs:{" & replaceInvisible($rhs) & "}\n"
-  when compiles(lhs.len):
+  when T is (string | seq | array):
```

Comment:
there could be other things with `len(lhs)` defined; 
that would be a good use case for https://github.com/nim-lang/Nim/pull/12076 which is much cheaper than 
`compiles` since it doesn't involve creating a new `PContext` and a `try/catch`; 

but fine i guess...

Reply:
This is a breaking change.

Reply:
It is not a breaking change, because it is not an exposed library.

Reply:
True I misread. Ok.